testing/security/transport_encryption.py

```diff
@@ -6,11 +6,63 @@
 
 
 import sdk_cmd
-
+import sdk_security
+import sdk_utils
 
 log = logging.getLogger(__name__)
 
 
+def setup_service_account(service_name: str,
+                          service_account_secret: str=None) -> dict:
+    """
+    Setup the service account for TLS
+    """
+
+    if sdk_utils.is_open_dcos():
+        log.warn("The setup of a service account requires DC/OS EE")
+        return {}
+
+    name = service_name
+    secret = name if service_account_secret is None else service_account_secret
+    sdk_security.create_service_account(service_account_name=name,
+                                        service_account_secret=secret)
+
+    if sdk_utils.dcos_version_less_than("1.11"):
+        sdk_cmd.run_cli("security org groups add_user superusers {name}".format(name=name))
+    else:
+        acls = [
+                {"rid": "dcos:secrets:default:/{}/*".format(service_name), "action": "full"},
+                {"rid": "dcos:secrets:list:default:/{}".format(service_name), "action": "read"},
+                {"rid": "dcos:adminrouter:ops:ca:rw", "action": "full"},
+                {"rid": "dcos:adminrouter:ops:ca:ro", "action": "full"},
+                ]
+
+        for acl in acls:
+            cmd_list = ["security", "org", "users", "grant",
+                        "--description", "\"Permissing required to provision TLS certificates\"",
+                        name, acl["rid"], acl["action"]
+                        ]
+
+            output = sdk_cmd.run_cli(" ".join(cmd_list))
+            log.info("output=%s", output)
+
+    return {"name": name, "secret": secret}
+
+
+def cleanup_service_account(service_account_info: dict):
```

Comment:
Won't this delete the service account in the middle of the strict suite?

Reply:
This is only called from within the `service_account` fixture for TLS tests (I didn't want to do a more large-scale refactor here). The only difference between the new behaviour and the existing behaviour is that we are not calling `delete_service_account` directly from the fixture.

What may be happening is that the `configure_security` fixture is using a different service account (`'service-acct'`) and such we have not picked it up until now. I will investigate further tomorrow and create a JIRA if applicable.

Reply:
Since we were using `superuser` before, it just "worked" in strict.

We'll need to make sure that whatever SA we use has the right permissions to run in strict.

Reply:
Yes, just thought of that as well ... will check tomorrow.